glmtools/glm.py

```diff
@@ -53,5 +61,33 @@ def t_test(c, X, B, sigma_2, df):
     p : array shape (V,)
         two-tailed probability value for each t statistic.
     """
-    # Your code code here
-    return
+
+    #get t statistic
+    c_b_cov = c.dot(npl.pinv(X.T.dot(X))).dot(c)
+    t = c.dot(B) / np.sqrt(sigma_2 * c_b_cov)
+
+    #get two-tailed p-value for t statistic
+    t_dist = scipy.stats.t(df=df)
+    if t > 0:
+        p = 2 * (1 - t_dist.cdf(t))
+    else:
+        p = 2 * (t_dist.cdf(t))
+    return t, p
+
+def test_glm_t_test():
+    """ needs to be tested with Y matrix that has 2 groups of 10
+    """
+    #generate fake data to test functions
+    n = 20
+    x = np.random.normal(10, 2, size=n)
+    X = np.ones((n, 2))
+    X[:, 1] = x
+    Y = np.random.normal(20, 1, size=n)
+
+    B, sigma_2, df = glm(Y,X)
+    c = np.array([0, 1])
+    t, p = t_test(c, X, B, sigma_2, df)
+
+    res = scipy.stats.linregress(x, Y)
+
+    return np.allclose(p, res.pvalue)
```

Comment:
You need to use ``assert`` to check the results are correct.  I believe py.test ignores the return values of the test functions.  In this case you need ``assert np.allclose(p, res.pvalue)``.  But you can also test the intercept and slope beta values here.